app/server/appsmith-git/src/main/resources/git.sh

```
@@ -50,6 +72,218 @@ git_download() {
+
+        echo "Searching in clone_and_checkout key: $key_value_pair_key" >&1
+
+        git_clone "$private_key" "$remote_url" "$target_folder"
+        git -C "$target_folder" config user.name "$author_name"
+        git -C "$target_folder" config user.email "$author_email"
+        git -C "$target_folder" config fetch.parallel 4
+
+
+        # This provides all the arguments from arg 5 onwards to the function git_co_from_redis.
+        # This includes the target folder, redis url, key value pair key, and all branch names from db.
+        git_co_from_redis ${@:5}
```

Shellcheck (0.10.0)

[error] Double quote array expansions to avoid re-splitting elements.

(SC2068)